In [1]:
import prepare_jpype
from test_train_loop import test_train_loop_MOA, test_train_loop_RIVER

MOA jar path location (config.ini): /Users/gomeshe/Dropbox/ciencia_computacao/dev/Using-MOA-API/moa.jar
JVM Location (system): 
/Users/gomeshe/Library/Java/JavaVirtualMachines/openjdk-20.0.1/Contents/Home
Sucessfully started the JVM and added MOA jar to the class path


In [2]:
## Datasets paths
csv_elec_tiny_path = '/Users/gomeshe/Desktop/data/electricity_tiny.csv'
arff_elec_tiny_path = '/Users/gomeshe/Desktop/data/electricity_tiny.arff'

## Basic classification examples

* Using the ```test_train_loop_MOA``` for MOA and ```test_train_loop_RIVER``` helper functions (see module test_train_loop)

* These are just examples on how to run several algorithms, it is kind of repetitive, but might be useful to have an example. 


## Examples using River

In [3]:
import pandas as pd
# Read the file to a dataframe, used by all examples
river_elec_tiny = pd.read_csv(csv_elec_tiny_path).to_numpy()

In [4]:
from river.forest import ARFClassifier

river_arf5 = ARFClassifier(
    n_models=5,
    max_features=0.60
)

acc, wallclock, cpu_time, df = test_train_loop_RIVER(river_arf5, river_elec_tiny)
print(f"{river_arf5}, {acc:.4f}, {wallclock:.4f}, {cpu_time:.4f}")

ARFClassifier, 0.8550, 0.3370, 0.3369


In [5]:
from river.tree import HoeffdingTreeClassifier

river_ht = HoeffdingTreeClassifier()

acc, wallclock, cpu_time, df = test_train_loop_RIVER(river_ht, river_elec_tiny)
print(f"{river_ht}, {acc:.4f}, {wallclock:.4f}, {cpu_time:.4f}")

HoeffdingTreeClassifier, 0.8230, 0.0644, 0.0600


In [6]:
from river.neighbors import KNNClassifier

river_knn = KNNClassifier()

acc, wallclock, cpu_time, df = test_train_loop_RIVER(river_knn, river_elec_tiny)
print(f"{river_knn}, {acc:.4f}, {wallclock:.4f}, {cpu_time:.4f}")

KNNClassifier, 0.7570, 2.0531, 2.0439


In [7]:
from river.naive_bayes import GaussianNB

river_gnb = GaussianNB()

acc, wallclock, cpu_time, df = test_train_loop_RIVER(river_gnb, river_elec_tiny)
print(f"{river_gnb}, {acc:.4f}, {wallclock:.4f}, {cpu_time:.4f}")

GaussianNB, 0.8490, 0.0285, 0.0285


In [8]:
from river.ensemble.streaming_random_patches import SRPClassifier

river_srp = SRPClassifier(
    n_models=5,
    subspace_size=0.6
)

acc, wallclock, cpu_time, df = test_train_loop_RIVER(river_srp, river_elec_tiny)
print(f"{river_srp}, {acc:.4f}, {wallclock:.4f}, {cpu_time:.4f}")

SRPClassifier(HoeffdingTreeClassifier), 0.8830, 0.7703, 0.7701


In [9]:
from river.tree import ExtremelyFastDecisionTreeClassifier

river_EFDT = ExtremelyFastDecisionTreeClassifier()

acc, wallclock, cpu_time, df = test_train_loop_RIVER(river_EFDT, river_elec_tiny)
print(f"{river_EFDT}, {acc:.4f}, {wallclock:.4f}, {cpu_time:.4f}")

ExtremelyFastDecisionTreeClassifier, 0.8230, 0.2326, 0.2334


## Examples using MOA

In [10]:
from moa.streams import ArffFileStream

elec_tiny_arff = ArffFileStream(arff_elec_tiny_path, -1)
elec_tiny_arff.prepareForUse()

In [11]:
from moa.classifiers.meta import AdaptiveRandomForest

arf10 = AdaptiveRandomForest()
arf10.getOptions().setViaCLIString("-s 5")
arf10.prepareForUse()

elec_tiny_arff.restart()

acc, wallclock, cpu_time, df = test_train_loop_MOA(elec_tiny_arff, arf10, maxInstances=2000, sampleFrequency=2000)
print(f"{arf10.getClass().getName()}, {acc:.4f}, {wallclock:.4f}, {cpu_time:.4f}")

moa.classifiers.meta.AdaptiveRandomForest, 89.4500, 0.5203, 1.0742


In [12]:
from moa.classifiers.trees import HoeffdingTree

ht_moa = HoeffdingTree()
ht_moa.prepareForUse()

elec_tiny_arff.restart()

acc, wallclock, cpu_time, df = test_train_loop_MOA(elec_tiny_arff, ht_moa, maxInstances=2000, sampleFrequency=2000)
print(f"{ht_moa.getClass().getName()}, {acc:.4f}, {wallclock:.4f}, {cpu_time:.4f}")

moa.classifiers.trees.HoeffdingTree, 86.0500, 0.0275, 0.0592


In [13]:
from moa.classifiers.lazy import kNN

knn_moa = kNN()
knn_moa.prepareForUse()

elec_tiny_arff.restart()

acc, wallclock, cpu_time, df = test_train_loop_MOA(elec_tiny_arff, knn_moa, maxInstances=2000, sampleFrequency=2000)
print(f"{knn_moa.getClass().getName()}, {acc:.4f}, {wallclock:.4f}, {cpu_time:.4f}")

moa.classifiers.lazy.kNN, 84.2167, 0.2595, 0.5288


In [14]:
from moa.classifiers.trees import EFDT

EFDT_moa = EFDT()
EFDT_moa.prepareForUse()

elec_tiny_arff.restart()

acc, wallclock, cpu_time, df = test_train_loop_MOA(elec_tiny_arff, EFDT_moa, maxInstances=2000, sampleFrequency=2000)
print(f"{EFDT_moa.getClass().getName()}, {acc:.4f}, {wallclock:.4f}, {cpu_time:.4f}")

moa.classifiers.trees.EFDT, 83.8375, 0.0253, 0.0508


In [15]:
from moa.classifiers.bayes import NaiveBayes

NaiveBayes_moa = NaiveBayes()
NaiveBayes_moa.prepareForUse()

elec_tiny_arff.restart()

acc, wallclock, cpu_time, df = test_train_loop_MOA(elec_tiny_arff, NaiveBayes_moa, maxInstances=2000, sampleFrequency=2000)
print(f"{NaiveBayes_moa.getClass().getName()}, {acc:.4f}, {wallclock:.4f}, {cpu_time:.4f}")

moa.classifiers.bayes.NaiveBayes, 83.8700, 0.0193, 0.0384


In [16]:
from moa.classifiers.meta import StreamingRandomPatches

SRP_moa = StreamingRandomPatches()
SRP_moa.getOptions().setViaCLIString("-s 10")
SRP_moa.prepareForUse()

elec_tiny_arff.restart()

acc, wallclock, cpu_time, df = test_train_loop_MOA(elec_tiny_arff, SRP_moa, maxInstances=2000, sampleFrequency=2000)
print(f"{SRP_moa.getClass().getName()}, {acc:.4f}, {wallclock:.4f}, {cpu_time:.4f}")

moa.classifiers.meta.StreamingRandomPatches, 84.9417, 0.5141, 1.0244


# Experiments MOA vs River using ARF

* These tests were using the old test_train_loop() and run_experiment() methods. 

* They are safe to delete or update (but the river ones take too long to execute).

* **SAVED THE RESULTS BECAUSE RIVER TAKES TOO LONG TO EXECUTE! Especially for ARF100!**

## MOA ARF10

In [48]:
%%time

from moa.classifiers.meta import AdaptiveRandomForest
from moa.core import Example
from moa.evaluation import BasicClassificationPerformanceEvaluator
from moa.streams import ArffFileStream

arf10 = AdaptiveRandomForest()
arf10.getOptions().setViaCLIString("-s 10")
arf10.setRandomSeed(1)
arf10.prepareForUse()

rtg_2abrupt = ArffFileStream("/Users/gomeshe/Desktop/data/RTG_2abrupt.arff", -1)
rtg_2abrupt.prepareForUse()

# Updated this one. 
test_train_loop_MOA(rtg_2abrupt, arf10, maxInstances=100000, sampleFrequency=1000)

CPU times: user 50.8 s, sys: 367 ms, total: 51.1 s
Wall time: 50.2 s


(83.88189655172414,
 50.219346046447754,
 51.13412499999999,
     classified instances  classifications correct (percent)  \
 0                17000.0                          86.317647   
 1                18000.0                          86.577778   
 2                19000.0                          86.794737   
 3                20000.0                          87.030000   
 4                21000.0                          87.266667   
 ..                   ...                                ...   
 95              112000.0                          84.234821   
 96              113000.0                          84.133628   
 97              114000.0                          84.052632   
 98              115000.0                          83.971304   
 99              116000.0                          83.881897   
 
     Kappa Statistic (percent)  Kappa Temporal Statistic (percent)  \
 0                   71.151605                            7.293743   
 1                   71.30485

CPU times: user 48.2 s, sys: 193 ms, total: 48.4 s
Wall time: 47.4 s

83.527000

## RIVER ARF10

In [4]:
%%time
from river.forest import ARFClassifier

rtg_2abrupt_path = '/Users/gomeshe/Desktop/data/RTG_2abrupt.csv'

river_arf10 = ARFClassifier(
    n_models=10,
    max_features=0.60
)

river_rtg_2abrupt = pd.read_csv(rtg_2abrupt_path).to_numpy()

result = run_experiment(river_arf10, river_rtg_2abrupt)
print('Learner: ', river_arf10, ' Result: ', result)

Learner:  ARFClassifier  Result:  (0.79017, 379)
CPU times: user 6min 17s, sys: 8.05 s, total: 6min 25s
Wall time: 6min 21s


CPU times: user 6min 5s, sys: 3.75 s, total: 6min 8s
Wall time: 6min 9s

Learner:  ARFClassifier  Result:  (0.80196, 368)

## MOA ARF30

In [28]:
%%time

from moa.classifiers.meta import AdaptiveRandomForest
from moa.core import Example
from moa.evaluation import BasicClassificationPerformanceEvaluator
from moa.streams import ArffFileStream

arf30 = AdaptiveRandomForest()
arf30.getOptions().setViaCLIString("-s 30")
arf30.setRandomSeed(1)
arf30.prepareForUse()

rtg_2abrupt = ArffFileStream("/Users/gomeshe/Desktop/data/RTG_2abrupt.arff", -1)
rtg_2abrupt.prepareForUse()

evaluator = BasicClassificationPerformanceEvaluator()
evaluator.recallPerClassOption.set()
evaluator.prepareForUse()

test_train_loop(rtg_2abrupt, arf30, evaluator, maxInstances=100000, sampleFrequency=1000)

CPU times: user 2min 40s, sys: 508 ms, total: 2min 40s
Wall time: 2min 36s


classified instances  classifications correct (percent)  \
0                 1000.0                          88.900000   
1                 2000.0                          89.550000   
2                 3000.0                          89.500000   
3                 4000.0                          90.050000   
4                 5000.0                          90.160000   
..                   ...                                ...   
95               96000.0                          84.034375   
96               97000.0                          83.921649   
97               98000.0                          83.835714   
98               99000.0                          83.754545   
99              100000.0                          83.666000   

    Kappa Statistic (percent)  Kappa Temporal Statistic (percent)  \
0                   47.856968                           58.426966   
1                   54.067347                           63.008850   
2                   55.650876                           62.941176   
3                   58.071433                           64.590747   
4                   58.288796                           64.806867   
..                        ...                                 ...   
95                  75.052102                           69.530644   
96                  74.852764                           69.387194   
97                  74.693075                           69.290865   
98                  74.535105                           69.184933   
99                  74.362581                           69.072009   

    Kappa M Statistic (percent)  Recall for class 0 (percent)  \
0                     31.901840                     98.685783   
1                     37.982196                     98.496693   
2                     39.539347                     98.305768   
3                     42.235123                     98.399275   
4                     42.523364                     98.455598   
..                          ...                           ...   
95                    68.703163                     96.978332   
96                    68.457245                     96.988813   
97                    68.253873                     97.006299   
98                    68.051251                     97.024250   
99                    67.828091                     97.034268   

    Recall for class 1 (percent)  Recall for class 2 (percent)  \
0                      38.650307                           NaN   
1                      45.400593                           NaN   
2                      47.600768                           NaN   
3                      49.927431                           NaN   
4                      50.000000                           NaN   
..                           ...                           ...   
95                     70.715536                     58.354866   
96                     70.620877                     58.025742   
97                     70.590061                     57.839428   
98                     70.551270                     57.618305   
99                     70.486585                     57.339922   

    Recall for class 3 (percent)  Recall for class 4 (percent)  
0                            NaN                           NaN  
1                            NaN                           NaN  
2                            NaN                           NaN  
3                            NaN                           NaN  
4                            NaN                           NaN  
..                           ...                           ...  
95                     17.525163                     90.082503  
96                     17.515924                     89.952536  
97                     17.362887                     89.764364  
98                     17.163555                     89.620224  
99                     17.144417                     89.467704  

[100 rows x 10 columns]

CPU times: user 2min 40s, sys: 508 ms, total: 2min 40s
Wall time: 2min 36s

83.666000

## RIVER ARF30

In [30]:
%%time

river_arf30 = ARFClassifier(
    n_models=30,
    max_features=0.60
)

river_rtg_2abrupt = pd.read_csv(rtg_2abrupt_path).to_numpy()

result = run_experiment(river_arf30, river_rtg_2abrupt)
print('Learner: ', river_arf30, ' Result: ', result)

Learner:  ARFClassifier  Result:  (0.80162, 1145)
CPU times: user 18min 56s, sys: 11.7 s, total: 19min 7s
Wall time: 19min 6s


CPU times: user 18min 56s, sys: 11.7 s, total: 19min 7s
Wall time: 19min 6s

Learner:  ARFClassifier  Result:  (0.80162, 1145)

## ARF100

In [31]:
%%time

from moa.classifiers.meta import AdaptiveRandomForest
from moa.core import Example
from moa.evaluation import BasicClassificationPerformanceEvaluator
from moa.streams import ArffFileStream

arf100 = AdaptiveRandomForest()
arf100.getOptions().setViaCLIString("-s 100")
arf100.setRandomSeed(1)
arf100.prepareForUse()

rtg_2abrupt = ArffFileStream("/Users/gomeshe/Desktop/data/RTG_2abrupt.arff", -1)
rtg_2abrupt.prepareForUse()

evaluator = BasicClassificationPerformanceEvaluator()
evaluator.recallPerClassOption.set()
evaluator.prepareForUse()

test_train_loop(rtg_2abrupt, arf100, evaluator, maxInstances=100000, sampleFrequency=1000)

CPU times: user 11min 15s, sys: 2.25 s, total: 11min 17s
Wall time: 10min 28s


classified instances  classifications correct (percent)  \
0                 1000.0                          88.900000   
1                 2000.0                          89.600000   
2                 3000.0                          89.466667   
3                 4000.0                          89.875000   
4                 5000.0                          90.120000   
..                   ...                                ...   
95               96000.0                          83.966667   
96               97000.0                          83.862887   
97               98000.0                          83.781633   
98               99000.0                          83.696970   
99              100000.0                          83.616000   

    Kappa Statistic (percent)  Kappa Temporal Statistic (percent)  \
0                   48.185077                           58.426966   
1                   54.086419                           63.185841   
2                   55.139083                           62.823529   
3                   56.827170                           63.967972   
4                   57.742296                           64.663805   
..                        ...                                 ...   
95                  74.936690                           69.401427   
96                  74.750844                           69.275311   
97                  74.598578                           69.188120   
98                  74.434459                           69.075720   
99                  74.274225                           68.977335   

    Kappa M Statistic (percent)  Recall for class 0 (percent)  \
0                     31.901840                     98.566308   
1                     38.278932                     98.616957   
2                     39.347409                     98.467124   
3                     41.219158                     98.550287   
4                     42.289720                     98.624517   
..                          ...                           ...   
95                    68.570437                     96.937929   
96                    68.341963                     96.961477   
97                    68.147658                     96.983431   
98                    67.938021                     97.003699   
99                    67.729609                     97.015987   

    Recall for class 1 (percent)  Recall for class 2 (percent)  \
0                      39.263804                           NaN   
1                      45.103858                           NaN   
2                      46.641075                           NaN   
3                      48.185776                           NaN   
4                      48.948598                           NaN   
..                           ...                           ...   
95                     70.651886                     58.936262   
96                     70.542352                     58.642698   
97                     70.496975                     58.448928   
98                     70.428652                     58.169527   
99                     70.335001                     57.978176   

    Recall for class 3 (percent)  Recall for class 4 (percent)  
0                            NaN                           NaN  
1                            NaN                           NaN  
2                            NaN                           NaN  
3                            NaN                           NaN  
4                            NaN                           NaN  
..                           ...                           ...  
95                     16.074600                     89.859058  
96                     16.184134                     89.718076  
97                     16.197783                     89.530439  
98                     16.160490                     89.392473  
99                     16.270816                     89.234790  

[100 rows x 10 columns]

CPU times: user 11min 15s, sys: 2.25 s, total: 11min 17s
Wall time: 10min 28s

83.616000

## RIVER ARF100

In [32]:
%%time

river_arf100 = ARFClassifier(
    n_models=100,
    max_features=0.60
)

river_rtg_2abrupt = pd.read_csv(rtg_2abrupt_path).to_numpy()

result = run_experiment(river_arf100, river_rtg_2abrupt)
print('Learner: ', river_arf100, ' Result: ', result)

Learner:  ARFClassifier  Result:  (0.80145, 4384)
CPU times: user 1h 13min 34s, sys: 42.7 s, total: 1h 14min 17s
Wall time: 2h 5s


CPU times: user 1h 13min 34s, sys: 42.7 s, total: 1h 14min 17s
Wall time: 2h 5s

Learner:  ARFClassifier  Result:  (0.80145, 4384)

## ARF100 -j 4

In [8]:
%%time

from moa.classifiers.meta import AdaptiveRandomForest
from moa.core import Example
from moa.evaluation import BasicClassificationPerformanceEvaluator
from moa.streams import ArffFileStream

arf100 = AdaptiveRandomForest()
arf100.getOptions().setViaCLIString("-s 100 -j 4")
arf100.setRandomSeed(1)
arf100.prepareForUse()

rtg_2abrupt = ArffFileStream("/Users/gomeshe/Desktop/data/RTG_2abrupt.arff", -1)
rtg_2abrupt.prepareForUse()

evaluator = BasicClassificationPerformanceEvaluator()
evaluator.recallPerClassOption.set()
evaluator.prepareForUse()

test_train_loop(rtg_2abrupt, arf100, evaluator, maxInstances=100000, sampleFrequency=1000)

CPU times: user 8min 25s, sys: 8.7 s, total: 8min 34s
Wall time: 3min 45s


classified instances  classifications correct (percent)  \
0                 1000.0                          88.900000   
1                 2000.0                          89.600000   
2                 3000.0                          89.466667   
3                 4000.0                          89.875000   
4                 5000.0                          90.120000   
..                   ...                                ...   
95               96000.0                          83.966667   
96               97000.0                          83.862887   
97               98000.0                          83.781633   
98               99000.0                          83.696970   
99              100000.0                          83.616000   

    Kappa Statistic (percent)  Kappa Temporal Statistic (percent)  \
0                   48.185077                           58.426966   
1                   54.086419                           63.185841   
2                   55.139083                           62.823529   
3                   56.827170                           63.967972   
4                   57.742296                           64.663805   
..                        ...                                 ...   
95                  74.936690                           69.401427   
96                  74.750844                           69.275311   
97                  74.598578                           69.188120   
98                  74.434459                           69.075720   
99                  74.274225                           68.977335   

    Kappa M Statistic (percent)  Recall for class 0 (percent)  \
0                     31.901840                     98.566308   
1                     38.278932                     98.616957   
2                     39.347409                     98.467124   
3                     41.219158                     98.550287   
4                     42.289720                     98.624517   
..                          ...                           ...   
95                    68.570437                     96.937929   
96                    68.341963                     96.961477   
97                    68.147658                     96.983431   
98                    67.938021                     97.003699   
99                    67.729609                     97.015987   

    Recall for class 1 (percent)  Recall for class 2 (percent)  \
0                      39.263804                           NaN   
1                      45.103858                           NaN   
2                      46.641075                           NaN   
3                      48.185776                           NaN   
4                      48.948598                           NaN   
..                           ...                           ...   
95                     70.651886                     58.936262   
96                     70.542352                     58.642698   
97                     70.496975                     58.448928   
98                     70.428652                     58.169527   
99                     70.335001                     57.978176   

    Recall for class 3 (percent)  Recall for class 4 (percent)  
0                            NaN                           NaN  
1                            NaN                           NaN  
2                            NaN                           NaN  
3                            NaN                           NaN  
4                            NaN                           NaN  
..                           ...                           ...  
95                     16.074600                     89.859058  
96                     16.184134                     89.718076  
97                     16.197783                     89.530439  
98                     16.160490                     89.392473  
99                     16.270816                     89.234790  

[100 rows x 10 columns]

## ARF100 -j 8

In [7]:
%%time

from moa.classifiers.meta import AdaptiveRandomForest
from moa.core import Example
from moa.evaluation import BasicClassificationPerformanceEvaluator
from moa.streams import ArffFileStream

arf100 = AdaptiveRandomForest()
arf100.getOptions().setViaCLIString("-s 100 -j 8")
arf100.setRandomSeed(1)
arf100.prepareForUse()

rtg_2abrupt = ArffFileStream("/Users/gomeshe/Desktop/data/RTG_2abrupt.arff", -1)
rtg_2abrupt.prepareForUse()

evaluator = BasicClassificationPerformanceEvaluator()
evaluator.recallPerClassOption.set()
evaluator.prepareForUse()

test_train_loop(rtg_2abrupt, arf100, evaluator, maxInstances=100000, sampleFrequency=1000)

CPU times: user 9min 43s, sys: 12.7 s, total: 9min 55s
Wall time: 4min 18s


classified instances  classifications correct (percent)  \
0                 1000.0                          88.900000   
1                 2000.0                          89.600000   
2                 3000.0                          89.466667   
3                 4000.0                          89.875000   
4                 5000.0                          90.120000   
..                   ...                                ...   
95               96000.0                          83.966667   
96               97000.0                          83.862887   
97               98000.0                          83.781633   
98               99000.0                          83.696970   
99              100000.0                          83.616000   

    Kappa Statistic (percent)  Kappa Temporal Statistic (percent)  \
0                   48.185077                           58.426966   
1                   54.086419                           63.185841   
2                   55.139083                           62.823529   
3                   56.827170                           63.967972   
4                   57.742296                           64.663805   
..                        ...                                 ...   
95                  74.936690                           69.401427   
96                  74.750844                           69.275311   
97                  74.598578                           69.188120   
98                  74.434459                           69.075720   
99                  74.274225                           68.977335   

    Kappa M Statistic (percent)  Recall for class 0 (percent)  \
0                     31.901840                     98.566308   
1                     38.278932                     98.616957   
2                     39.347409                     98.467124   
3                     41.219158                     98.550287   
4                     42.289720                     98.624517   
..                          ...                           ...   
95                    68.570437                     96.937929   
96                    68.341963                     96.961477   
97                    68.147658                     96.983431   
98                    67.938021                     97.003699   
99                    67.729609                     97.015987   

    Recall for class 1 (percent)  Recall for class 2 (percent)  \
0                      39.263804                           NaN   
1                      45.103858                           NaN   
2                      46.641075                           NaN   
3                      48.185776                           NaN   
4                      48.948598                           NaN   
..                           ...                           ...   
95                     70.651886                     58.936262   
96                     70.542352                     58.642698   
97                     70.496975                     58.448928   
98                     70.428652                     58.169527   
99                     70.335001                     57.978176   

    Recall for class 3 (percent)  Recall for class 4 (percent)  
0                            NaN                           NaN  
1                            NaN                           NaN  
2                            NaN                           NaN  
3                            NaN                           NaN  
4                            NaN                           NaN  
..                           ...                           ...  
95                     16.074600                     89.859058  
96                     16.184134                     89.718076  
97                     16.197783                     89.530439  
98                     16.160490                     89.392473  
99                     16.270816                     89.234790  

[100 rows x 10 columns]

CPU times: user 9min 43s, sys: 12.7 s, total: 9min 55s
Wall time: 4min 18s

83.616000	

## ARF100 -j 2

In [8]:
%%time

from moa.classifiers.meta import AdaptiveRandomForest
from moa.core import Example
from moa.evaluation import BasicClassificationPerformanceEvaluator
from moa.streams import ArffFileStream

arf100 = AdaptiveRandomForest()
arf100.getOptions().setViaCLIString("-s 100 -j 2")
arf100.setRandomSeed(1)
arf100.prepareForUse()

rtg_2abrupt = ArffFileStream("/Users/gomeshe/Desktop/data/RTG_2abrupt.arff", -1)
rtg_2abrupt.prepareForUse()

evaluator = BasicClassificationPerformanceEvaluator()
evaluator.recallPerClassOption.set()
evaluator.prepareForUse()

test_train_loop(rtg_2abrupt, arf100, evaluator, maxInstances=100000, sampleFrequency=1000)

CPU times: user 9min 42s, sys: 8.26 s, total: 9min 50s
Wall time: 6min 20s


classified instances  classifications correct (percent)  \
0                 1000.0                          88.900000   
1                 2000.0                          89.600000   
2                 3000.0                          89.466667   
3                 4000.0                          89.875000   
4                 5000.0                          90.120000   
..                   ...                                ...   
95               96000.0                          83.966667   
96               97000.0                          83.862887   
97               98000.0                          83.781633   
98               99000.0                          83.696970   
99              100000.0                          83.616000   

    Kappa Statistic (percent)  Kappa Temporal Statistic (percent)  \
0                   48.185077                           58.426966   
1                   54.086419                           63.185841   
2                   55.139083                           62.823529   
3                   56.827170                           63.967972   
4                   57.742296                           64.663805   
..                        ...                                 ...   
95                  74.936690                           69.401427   
96                  74.750844                           69.275311   
97                  74.598578                           69.188120   
98                  74.434459                           69.075720   
99                  74.274225                           68.977335   

    Kappa M Statistic (percent)  Recall for class 0 (percent)  \
0                     31.901840                     98.566308   
1                     38.278932                     98.616957   
2                     39.347409                     98.467124   
3                     41.219158                     98.550287   
4                     42.289720                     98.624517   
..                          ...                           ...   
95                    68.570437                     96.937929   
96                    68.341963                     96.961477   
97                    68.147658                     96.983431   
98                    67.938021                     97.003699   
99                    67.729609                     97.015987   

    Recall for class 1 (percent)  Recall for class 2 (percent)  \
0                      39.263804                           NaN   
1                      45.103858                           NaN   
2                      46.641075                           NaN   
3                      48.185776                           NaN   
4                      48.948598                           NaN   
..                           ...                           ...   
95                     70.651886                     58.936262   
96                     70.542352                     58.642698   
97                     70.496975                     58.448928   
98                     70.428652                     58.169527   
99                     70.335001                     57.978176   

    Recall for class 3 (percent)  Recall for class 4 (percent)  
0                            NaN                           NaN  
1                            NaN                           NaN  
2                            NaN                           NaN  
3                            NaN                           NaN  
4                            NaN                           NaN  
..                           ...                           ...  
95                     16.074600                     89.859058  
96                     16.184134                     89.718076  
97                     16.197783                     89.530439  
98                     16.160490                     89.392473  
99                     16.270816                     89.234790  

[100 rows x 10 columns]

CPU times: user 9min 42s, sys: 8.26 s, total: 9min 50s
Wall time: 6min 20s

83.616000

## ARF100 -j 5

In [9]:
%%time

from moa.classifiers.meta import AdaptiveRandomForest
from moa.core import Example
from moa.evaluation import BasicClassificationPerformanceEvaluator
from moa.streams import ArffFileStream

arf100 = AdaptiveRandomForest()
arf100.getOptions().setViaCLIString("-s 100 -j 5")
arf100.setRandomSeed(1)
arf100.prepareForUse()

rtg_2abrupt = ArffFileStream("/Users/gomeshe/Desktop/data/RTG_2abrupt.arff", -1)
rtg_2abrupt.prepareForUse()

evaluator = BasicClassificationPerformanceEvaluator()
evaluator.recallPerClassOption.set()
evaluator.prepareForUse()

test_train_loop(rtg_2abrupt, arf100, evaluator, maxInstances=100000, sampleFrequency=1000)

CPU times: user 9min 39s, sys: 8.91 s, total: 9min 48s
Wall time: 4min 37s


classified instances  classifications correct (percent)  \
0                 1000.0                          88.900000   
1                 2000.0                          89.600000   
2                 3000.0                          89.466667   
3                 4000.0                          89.875000   
4                 5000.0                          90.120000   
..                   ...                                ...   
95               96000.0                          83.966667   
96               97000.0                          83.862887   
97               98000.0                          83.781633   
98               99000.0                          83.696970   
99              100000.0                          83.616000   

    Kappa Statistic (percent)  Kappa Temporal Statistic (percent)  \
0                   48.185077                           58.426966   
1                   54.086419                           63.185841   
2                   55.139083                           62.823529   
3                   56.827170                           63.967972   
4                   57.742296                           64.663805   
..                        ...                                 ...   
95                  74.936690                           69.401427   
96                  74.750844                           69.275311   
97                  74.598578                           69.188120   
98                  74.434459                           69.075720   
99                  74.274225                           68.977335   

    Kappa M Statistic (percent)  Recall for class 0 (percent)  \
0                     31.901840                     98.566308   
1                     38.278932                     98.616957   
2                     39.347409                     98.467124   
3                     41.219158                     98.550287   
4                     42.289720                     98.624517   
..                          ...                           ...   
95                    68.570437                     96.937929   
96                    68.341963                     96.961477   
97                    68.147658                     96.983431   
98                    67.938021                     97.003699   
99                    67.729609                     97.015987   

    Recall for class 1 (percent)  Recall for class 2 (percent)  \
0                      39.263804                           NaN   
1                      45.103858                           NaN   
2                      46.641075                           NaN   
3                      48.185776                           NaN   
4                      48.948598                           NaN   
..                           ...                           ...   
95                     70.651886                     58.936262   
96                     70.542352                     58.642698   
97                     70.496975                     58.448928   
98                     70.428652                     58.169527   
99                     70.335001                     57.978176   

    Recall for class 3 (percent)  Recall for class 4 (percent)  
0                            NaN                           NaN  
1                            NaN                           NaN  
2                            NaN                           NaN  
3                            NaN                           NaN  
4                            NaN                           NaN  
..                           ...                           ...  
95                     16.074600                     89.859058  
96                     16.184134                     89.718076  
97                     16.197783                     89.530439  
98                     16.160490                     89.392473  
99                     16.270816                     89.234790  

[100 rows x 10 columns]